src/routes/InstancesPage/InstancesToolbarSearchFilter.js

```diff
@@ -0,0 +1,230 @@
+import SelectSingle from '../../components/SelectSingle';
+
+function InstancesToolbarSearchFilter({ filters, setFilters }) {
+  const [selectedFilter, setSelectedFilter] = useState('Name');
+  // local state for input values
+  const [inputName, setInputName] = useState('');
+  const [inputOwner, setInputOwner] = useState('');
+  // local state for Select isExpanded values
+  // @TODO: We can refactor the SelectSingle component to be more reusable for the usecase in this component as well. Then we don't need to keep this state here.
+  const [isRegionExpanded, setIsRegionExpanded] = useState(false);
+  const [isStatusExpanded, setIsStatusExpanded] = useState(false);
+
+  // TODO: Extract into separate utils file to be reused in other cases
+  function onDeleteChip(type = '', id = '') {
+    if (type) {
+      setFilters((prevFilters) => {
+        const newFilters = { ...prevFilters };
+        newFilters[type.toLowerCase()] = newFilters[type.toLowerCase()].filter(
+          (s) => s !== id
+        );
+        return newFilters;
+      });
+    } else {
+      setFilters({
+        region: [],
+      });
+    }
+  }
+
+  // TODO: Extract into separate utils file to be reused in other cases
+  function onDeleteChipGroup(type) {
+    setFilters((prevFilters) => {
+      const newFilters = { ...prevFilters };
+      newFilters[type.toLowerCase()] = [];
+      return newFilters;
+    });
+  }
+
+  // TODO: Extract into separate utils file to be reused in other cases
+  function onSelect(type, event, selection) {
+    const checked = event.target.checked;
+    setFilters((prevFilters) => {
+      const prevSelections = prevFilters[type] || [];
+      return {
+        ...prevFilters,
+        [type]: checked
+          ? [...prevSelections, selection]
+          : prevSelections.filter((value) => value !== selection),
+      };
+    });
+  }
+
+  function onRegionSelect(event, selection) {
+    onSelect('region', event, selection);
+  }
+
+  function onStatusSelect(event, selection) {
+    onSelect('status', event, selection);
+  }
+
+  return (
+    <ToolbarToggleGroup toggleIcon={<FilterIcon />} breakpoint="xl">
+      <ToolbarGroup variant="filter-group">
+        <ToolbarItem>
+          <SelectSingle
+            id="region"
+            value={selectedFilter}
+            handleSelect={(_, selection) => {
+              setSelectedFilter(selection);
+            }}
+          >
+            <SelectOption value="Name">Name</SelectOption>
+            <SelectOption value="Region">Region</SelectOption>
+            <SelectOption value="Owner">Owner</SelectOption>
+            <SelectOption value="Status">Status</SelectOption>
+          </SelectSingle>
+        </ToolbarItem>
+        <ToolbarFilter
+          chips={filters.name}
+          deleteChip={onDeleteChip}
+          deleteChipGroup={onDeleteChipGroup}
+          categoryName="Name"
+          className={selectedFilter !== 'Name' && 'pf-u-hidden'}
```

Comment:
Conditionally rendering the `ToolbarFilter` actually affects the filter so we'll need to actually keep it rendered and conditionally hide/show it instead. I'm using the `pf-u-hidden` class but other suggestions are welcome

Reply:
This seems like it is probably the best way given how these function. If we wanted to get really strict we could create a wrapper around `ToolbarFilter` to ensure that the `categoryName` prop matches the comparison in `className`, but that's probably overkill. I image any typos would be almost immediately noticeable when new filters are added.